lib/mayaUsd/undo/ForwardDeclares.h

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef MAYAUSD_UNDO_FORWARD_DECLARES_H
+#define MAYAUSD_UNDO_FORWARD_DECLARES_H
+
+#include <pxr/pxr.h>
+#include <pxr/usd/usd/prim.h>
```

Comment:
@ppt-adsk Yup, this is a long story. prim.h brings in a lot of headers and unfortunately it is common pattern in many of the codes around.  I was planning to clean it up but it turn out to be a lot more work than I expected.

For now, I removed the ForwardDeclares.h since it was a bit deceiving and clean up the header inclusions a little bit.

see 026a964e06a51e4a5d8d386e9a7cc50eb4e98355